src/main/antlr4/liquid/parser/v4/LiquidParser.g4

```diff
@@ -250,8 +274,8 @@ other_than_tag_end
  : ~TagEnd+
  ;
 
-other_than_tag_end_out_start
- : ~(TagEnd | OutStart | OutStart2)+
+filename
+ : ( . )+?
```

Comment:
I'm no fan of `.*?` or `.+? ` in productions (parser rules) that do not have a clear end. With a "clear end" I mean this:

```
filename
 : .+? SomeToken
 ;
```

where it is clear the `.+?` ends when `SomeToken` is encountered. Without the "clear end" it is hard to reason when the `.+?` will stop matching. Is it not possible to keep the `~(TagEnd | OutStart | OutStart2)+`, or extend the negated set `~(TagEnd | OutStart | OutStart2 | OtherTokenA )+`?